doc/release-notes/rl-2505.section.md

```diff
@@ -25,9 +25,12 @@
   The `nixLog` function, which logs unconditionally, was also re-introduced and modified to prefix messages with the function name of the caller.
   For more information, [see this PR](https://github.com/NixOS/nixpkgs/pull/370742).
 
-- The `rustPlatform.fetchCargoTarball` function is deprecated, because it relied on `cargo vendor` not changing its output format to keep fixed-output derivation hashes the same, which is a Nix invariant, and Cargo 1.84.0 changed `cargo vendor`'s output format.
-  It should generally be replaced with `rustPlatform.fetchCargoVendor`, but `rustPlatform.importCargoLock` may also be appropriate in some circumstances.
-  `rustPlatform.buildRustPackage` users must set `useFetchCargoVendor` to `true` and regenerate the `cargoHash`.
+- Rust packages will need to regenerate their `cargoHash`.
+  Cargo 1.84.0 changed the format of `cargo vendor` output, which invalidated all existing `rustPlatform.fetchCargoTarball` hashes.
```

Comment:
Let's clear up the terminology here:

To me, when something "invalidates" a hash it doesn't mean that it causes a hash mismatch, it just means that it will try to re-check the hash, thus re-fetching the files. Unrelated whether or not there's a hash mismatch or not.

Here, I assume you meant something like "broke all existing hashes" (not sure what else to call it other than "break"), meaning those FODs will cause a hash mismatch if they are re-fetched. However they are very much not invalidated, in the sense that they will still be fetched from the cache (because the name didn't change)

Reply:
In a literal sense, the hashes were previously valid and are no longer. I agree that there could potentially be confusion with the specific sense of cache invalidation (which is an operation you perform when the underlying result changes, making the cache entry no longer valid), but I’m hoping the context here makes it clear?

I don’t object to “broke” here if you have a strong opinion, though.

Reply:
There is a helper called `invalidateFetcherByDrvHash` that makes the `name` value of the FOD always dependent on the derivation's hash, thus always "invalidating" it (making it always get re-fetched).
However I couldn't find much in favor of my definition being the correct usage.

I see that in https://github.com/NixOS/nixpkgs/pull/376038  "invalidate" also meant "break".

So... I don't know. Let's keep it as "invalidate".

Though, if the "invalidate" naming is already taken what would you call the "needs-to-be-refetched-to-make-sure-its-actually-correct" state of a FOD?

Reply:
I guess it’s refreshing a cache entry, because you think it might be in need of cache invalidation, because perhaps the recorded hash it’s keyed on was invalidated :)